ksql-serde/src/main/java/io/confluent/ksql/serde/Format.java

```diff
@@ -15,69 +15,29 @@
 
 package io.confluent.ksql.serde;
 
-import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Sets;
 import com.google.common.collect.Sets.SetView;
-import io.confluent.ksql.serde.avro.KsqlAvroSerdeFactory;
-import io.confluent.ksql.serde.delimited.KsqlDelimitedSerdeFactory;
-import io.confluent.ksql.serde.json.KsqlJsonSerdeFactory;
-import io.confluent.ksql.serde.kafka.KafkaSerdeFactory;
-import io.confluent.ksql.util.KsqlConstants;
 import io.confluent.ksql.util.KsqlException;
 import java.util.Map;
-import java.util.Optional;
 import java.util.Set;
 
-public enum Format {
-
-  JSON(true, ImmutableSet.of(), ImmutableSet.of()) {
-    @Override
-    public KsqlSerdeFactory getSerdeFactory(final FormatInfo info) {
-      return new KsqlJsonSerdeFactory();
-    }
-  },
-
-  AVRO(true, ImmutableSet.of(FormatInfo.FULL_SCHEMA_NAME), ImmutableSet.of()) {
-    @Override
-    public KsqlSerdeFactory getSerdeFactory(final FormatInfo info) {
-      final String schemaFullName = info.getProperties()
-          .getOrDefault(FormatInfo.FULL_SCHEMA_NAME, KsqlConstants.DEFAULT_AVRO_SCHEMA_FULL_NAME);
-
-      return new KsqlAvroSerdeFactory(schemaFullName);
-    }
-  },
-
-  DELIMITED(false, ImmutableSet.of(FormatInfo.DELIMITER), ImmutableSet.of(FormatInfo.DELIMITER)) {
-    @Override
-    public KsqlSerdeFactory getSerdeFactory(final FormatInfo info) {
-      return new KsqlDelimitedSerdeFactory(
-          Optional.ofNullable(
-              info.getProperties().get(FormatInfo.DELIMITER)
-          ).map(Delimiter::parse)
-      );
-    }
-  },
-
-  KAFKA(false, ImmutableSet.of(), ImmutableSet.of()) {
-    @Override
-    public KsqlSerdeFactory getSerdeFactory(final FormatInfo info) {
-      return new KafkaSerdeFactory();
-    }
-  };
-
-  private final boolean supportsUnwrapping;
-  private final Set<String> validConfigs;
-  private final Set<String> inheritableProperties;
+/**
+ * A {@code Format} is a serialization specification of a Kafka topic
+ * in ksqlDB. The builtin formats are specified in the {@link Formats}
+ * class.
+ */
+public interface Format {
```

Comment:
Can we add a `@PublicApi` annotation and mark this with it, (and any others you can think of, e.g. UDF family of stuff).

(Separate PR, obvs).

Reply:
https://github.com/confluentinc/ksql/issues/4439